galter_subjects_utils/updater.py

```diff
@@ -269,36 +293,68 @@ def _update_rdm_records(self):
             )
 
     def _remove_rdm_subjects(self):
-        """Remove subjects from the Subjects entries."""
+        """Remove subjects from the Subjects entries.
+
+        We have to resort to low-level commands because the high-level ones
+        are not made for bulk operations. We've checked the implications
+        and we should be fine (at least at time of writing).
+        """
+        ids_for_removal = [
+            op["id"] for op in self._ops_data
+            if op.get("type") in ["remove", "replace"]
+        ]
         service = current_service_registry.get("subjects")
 
-        remove_ops = filter_ops_by_type(self._ops_data, "remove")
-        replace_ops = filter_ops_by_type(self._ops_data, "replace")
-        for op in remove_ops + replace_ops:
-            try:
-                service.delete(
-                    system_identity,
-                    op["id"]
-                )
-            except PersistentIdentifierError:
-                # For any persistent identifier related problem
-                # we just ignore it and make sure we can continue.
-                # Main scenario is when a subject has already been
-                # "service.delete"'d in which case its PID is only marked as
-                # deleted which interferes with service.delete and
-                # service.create.
-                pass
-
-            # Purge (completely delete) backing subject PID
-            # This operation is idempotent.
-            delete_pid_stmt = (
+        model_cls = service.record_cls.model_cls
+        size_of_batch = 200  # Maybe TODO: make configurable
+        for offset in range(0, len(ids_for_removal), size_of_batch):
+            batch = ids_for_removal[offset:offset + size_of_batch]
```

Comment:
note: python allows array slicing larger than array size and does the right thing: gives only part that does exist or `[]` if none at all.

Reply:
I'm really surprised by that, but that's nice to know 👍